File: Leet 2075/solution.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        if not encodedText:
            return ''

        length: int = len(encodedText)
        cols: int = length // rows
        
        # build matrix
        matrix: list[list[str]] = []
        for i in range(0, length, cols):
            matrix.append([c for c in encodedText[i:i+cols]])

        # decode
        builder: list[str] = []
        for i in range(cols):
            for r in range(rows):
                if i >= cols:
                    break
                builder.append(matrix[r][i])
                i += 1

        return ''.join(builder).rstrip()
    

def main() -> None:
    print('2075. Decode the Slanted Ciphertext')

    solution = Solution()
    case1_result = solution.decodeCiphertext('ch   ie   pr', 3)
    print(case1_result)


if __name__ == '__main__':
    main()

File: Leet 2075/solution.cs
using System.Text;

public class Solution
{
    public string DescodeCiphertext(string encodedText, int rows, bool debug = false)
    {
        if(encodedText == "")
        {
            return "";
        }

        var cols = encodedText.Length / rows;
        char[,] matrix = new char[rows, cols];

        for(int i = 0; i < encodedText.Length; i++)
        {
            var row = i / cols;
            var col = i % cols;
            matrix[row, col] = encodedText[i];
        }

        if (debug)
        {
            for(int i = 0; i < cols; i++)
            {
                Console.Write("[ ");
                for(int j = 0; j < rows; j++)
                {
                    Console.Write($" {matrix[j, i]} ");
                }
                Console.WriteLine(" ]");
            }
        }

        StringBuilder stringBuilder = new();        
        for(int col = 0; col < cols; col++)
        {
            var k = col;
            for(int row = 0; row < rows; row++)
            {
                if (k >= cols) 
                {
                    break;
                }
                
                stringBuilder.Append(matrix[row,k]);

                k++;
            }
        }

        return stringBuilder.ToString().TrimEnd();

    }
}


class Program
{
    private static void Main()
    {
        Solution solution = new();
        var text1 = solution.DescodeCiphertext("ch   ie   pr", 3);
        Console.WriteLine($"Result: {text1} {text1.Length}");

        var text2 = solution.DescodeCiphertext("iveo    eed   l te   olc", 4);
        Console.WriteLine($"Result: {text2} {text2.Length}");
    }
}